src/resources/ha-style.html

```diff
@@ -5,53 +5,37 @@
   body {
     font-size: 14px;
 
-    /* for paper-toggle-button */
-    --paper-grey-50: #fafafa;
-    --paper-grey-200: #eeeeee;
-
-    --paper-item-icon-color: #44739e;
-    --paper-item-icon-active-color: #FDD835;
-
+    --primary-color: #03A9F4; /* header */
     --dark-primary-color: #0288D1;
-    --primary-color: #03A9F4;
     --light-primary-color: #B3E5FC;
-    --text-primary-color: #ffffff;
-    --accent-color: #FF9800;
-    --primary-background-color: var(--paper-grey-50);
-    --secondary-background-color: #E5E5E5;
+    --text-primary-color: #FFFFFF; /* header */
+    --primary-background-color: #fafafa;
     --primary-text-color: #212121;
+
+    --secondary-background-color: #E5E5E5; /* states, */
     --secondary-text-color: #727272;
+
+    --tertiary-background-color: #FFFFFF; /* badges, state cards, side menu */
+    --tertiary-text-color: rgb(76, 76, 76); /* badges, */
+
     --disabled-text-color: #bdbdbd;
     --divider-color: rgba(0, 0, 0, .12);
+    --accent-color: #FF9800;
 
     --table-row-background-color: transparant;
-    --table-row-alternative-background-color: #eee;
+    --table-row-alternative-background-color: var(--secondary-background-color);
 
-    --paper-toggle-button-checked-ink-color: #039be5;
-    --paper-toggle-button-checked-button-color: #039be5;
-    --paper-toggle-button-checked-bar-color: #039be5;
-
-    --paper-slider-knob-color: var(--primary-color);
-    --paper-slider-knob-start-color: var(--primary-color);
-    --paper-slider-pin-color: var(--primary-color);
-    --paper-slider-active-color: var(--primary-color);
-    --paper-slider-secondary-color: var(--light-primary-color);
-    --paper-slider-container-color: var(--divider-color);
+    --paper-item-icon-color: #44739e;
+    --paper-item-icon-active-color: #FDD835;
 
-    --paper-card-background-color: #FFF;
-    --paper-listbox-background-color: #FFF;
+    --sidebar-text_-_background: var(--tertiary-background-color);
 
-    /* for label-badge */
-    --label-badge-background-color: white;
-    --label-badge-text-color: rgb(76, 76, 76);
-    --label-badge-red: #DF4C1E;
-    --label-badge-blue: #039be5;
-    --label-badge-green: #0DA035;
-    --label-badge-yellow: #f4b400;
-    --label-badge-grey: var(--paper-grey-500);
+    /* for paper-toggle-button */
+    --paper-grey-50: var(--primary-background-color);
```

Comment:
We should not redefine paper-greys. That's an official Material Design color.

When defining styling, there is a hierarchy to things.

1. Define color names (ie paper-grey-50)
2. Define brand colors using color names (ie primary color, accent color)
3. Define specific applications of the brand colors to UI elements using brand colors. Never use color names directly. (--paper-slider-container-color)

If we follow this correctly, theming can focus on the variables defined in layer 2.